velox/functions/prestosql/ArrayIntersectExcept.cpp

```diff
@@ -348,33 +448,68 @@ std::shared_ptr<exec::VectorFunction> createArrayIntersect(
 std::shared_ptr<exec::VectorFunction> createArrayExcept(
     const std::string& name,
     const std::vector<exec::VectorFunctionArg>& inputArgs) {
-  validateType(inputArgs);
+  validateType(inputArgs, name, 2);
   auto elementType = inputArgs.front().type->childAt(0);
 
   return VELOX_DYNAMIC_SCALAR_TEMPLATE_TYPE_DISPATCH(
-      createTyped, /* isIntersect */ false, elementType->kind(), inputArgs);
+      createTypedArraysIntersectExcept,
+      /* isIntersect */ false,
+      elementType->kind(),
+      inputArgs);
 }
 
-std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
+std::vector<std::shared_ptr<exec::FunctionSignature>> signatures(
+    const std::string returnType) {
   // array(T), array(T) -> array(T)
   return {exec::FunctionSignatureBuilder()
               .typeVariable("T")
-              .returnType("array(T)")
+              .returnType(returnType)
               .argumentType("array(T)")
               .argumentType("array(T)")
               .build()};
 }
 
+template <TypeKind kind>
+const std::shared_ptr<exec::VectorFunction> createTypedArraysOverlap(
+    const std::vector<exec::VectorFunctionArg>& inputArgs) {
+  VELOX_CHECK_EQ(inputArgs.size(), 2);
+  BaseVector* left = inputArgs[0].constantValue.get();
+  BaseVector* right = inputArgs[1].constantValue.get();
+  using T = typename TypeTraits<kind>::NativeType;
+  const bool isLeftConstant = (left != nullptr);
```

Comment:
move this variable closer to where it is used, e.g. after the `if (left == nullptr && right == nullptr)` statement.